### ISSUE TITLE
When declaring a class, I cannot refer to the class in a contract...?

### ISSUE THREAD

**Issue by beetleb** [AndreaCensi/contracts#34]
The following code doesn't work:

```
class CommodityType(object):
    """
    A generic commodity type.
    """
    def __init__(self, symbol):
	self.symbol = symbol

    # @contract(other=CommodityType)
    def __eq__(self, other):
	return self.symbol == other.symbol
```

It complains that CommodityType is not defined. Is there any way around this problem? I was trying to avoid having an isinstance check inside.

Thanks,

**Comment by AndreaCensi** [comment]
No, this is a limitation of Python. You cannot refer to the class being
defined.



On Saturday, February 14, 2015, beetleb <notifications@github.com> wrote:

> The following code doesn't work:
>
> class CommodityType(object):
>     """
>     A generic commodity type.
>     """
>     def __init__(self, symbol):
>     self.symbol = symbol
>
>     # @contract(other=CommodityType)
>     def __eq__(self, other):
>     return self.symbol == other.symbol
>
> It complains that CommodityType is not defined. Is there any way around
> this problem? I was trying to avoid having an isinstance check inside.
>
> Thanks,
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/AndreaCensi/contracts/issues/34>.
>


-- 
(sent from mobile device, please pardon typos and brevity)

**Comment by AndreaCensi** [comment]
I'm leaving this open, because we need to find some workaround, as this is a very common case.

**Comment by AndreaCensi** [comment]
I'm still not sure about this could be implemented. It seems it is a limitation of Python and the workaround would be very complicated. We would need to defer the evaluation of the contracts.

**Comment by timClicks** [comment]
I believe SQLAlchemy shares this issue and they resolve it by using strings.
